mne_connectivity/base.py

```diff
@@ -0,0 +1,525 @@
+import numpy as np
+import xarray as xr
+
+from mne.utils import sizeof_fmt, object_size, _validate_type
+from mne_connectivity.utils import fill_doc
+
+
+class SpectralMixin:
+    @property
+    def freqs(self):
+        return self.xarray.coords.get('freqs').values.tolist()
+
+
+class TimeMixin:
+    @property
+    def times(self):
+        return self.xarray.coords.get('times').values.tolist()
+
+
+@fill_doc
+class _Connectivity():
+    # whether or not the connectivity occurs over epochs
+    is_epoched = False
+
+    def __init__(self, data, names, indices, method,
+                 n_nodes, **kwargs):
+        """Base class container for connectivity data.
+
+        Connectivity data is anything that represents "connections"
+        between nodes as a (N, N) array. It can be symmetric, or
+        asymmetric (if it is symmetric, storage optimization will
+        occur).
+
+        The underlying data structure is an ``xarray.DataArray``,
+        with a similar API. We provide support for storing
+        connectivity data in a subset of nodes. Thus the underlying
+        data structure instead of a ``(n_nodes_in, n_nodes_out)`` 2D array
+        would be a ``(n_nodes_in * n_nodes_out,)`` raveled 1D array. This
+        allows us to optimize storage also for symmetric connectivity.
```

Comment:
Should this go into the `Notes` section?

Reply:
Good point! Done.